cli/tests/integrations/test_marathon.py

```diff
@@ -687,3 +702,29 @@ def _zero_instance_app():
     with app('tests/data/marathon/apps/zero_instance_sleep.json',
              'zero-instance-app'):
         yield
+
+
+@contextlib.contextmanager
+def _zero_instance_app_through_http():
+    class JSONRequestHandler (BaseHTTPRequestHandler):
+
+        def do_GET(self):
+            self.send_response(200)
+            self.send_header("Content-type", "application/json")
+            self.end_headers()
+            self.wfile.write(open(
+                'tests/data/marathon/apps/zero_instance_sleep.json',
+                'rb').read())
+
+    host = 'localhost'
+    port = 12345
+    server = HTTPServer((host, port), JSONRequestHandler)
+    thread = threading.Thread(target=server.serve_forever)
+    thread.setDaemon(True)
+    thread.start()
+
```

Comment:
Why do you need to set up all this? `dcos marathon app add urlhere` should just work. See [alternative pr](https://github.com/mesosphere/dcos-cli/pull/246/files).